src/shared/components/ShareDropdown/ShareDropdown.tsx

```diff
@@ -81,25 +84,43 @@ const ShareDropdown: FC<ShareDropdownProps> = ({
 
 			<DropdownContent>
 				<div className="c-share-dropdown__container">
-					<Tabs tabs={tabs} onClick={(id) => setActiveTab(id)} />
+					{withPupils ? (
+						<>
+							<Tabs tabs={tabs} onClick={(id) => setActiveTab(id)} />
 
-					<div className="c-share-dropdown__content">
-						{tab === ShareDropdownTabs.COLLEAGUES ? (
-							<>
-								{users && (
-									<ShareWithColleagues
-										contributors={users}
-										onAddNewContributor={onAddNewUser}
-										onDeleteContributor={onDeleteUser}
-										onEditRights={onEditRights}
-										hasModalOpen={(open: boolean) => setHasModalOpen(open)}
-									/>
+							<div className="c-share-dropdown__content">
+								{tab === ShareDropdownTabs.COLLEAGUES ? (
+									<>
+										{contributors && (
+											<ShareWithColleagues
+												contributors={contributors}
+												onAddNewContributor={onAddContributor}
+												onDeleteContributor={onDeleteContributor}
+												onEditRights={onEditContributorRights}
+												hasModalOpen={(open: boolean) =>
+													setHasModalOpen(open)
+												}
+											/>
```

Comment:
extract into a render function:
```suggestion
											{renderShareWithCollegues()}
```

and reuse the render function below